packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -1,175 +1,181 @@
-    } else if (e.code == 'AmplifyAlreadyConfiguredException') {
-      return AmplifyAlreadyConfiguredException.fromMap(
-          Map<String, String>.from(e.details));
-    } else {
-      // This shouldn't happen. All exceptions coming from platform for
-      // amplify_datastore should have a known code. Throw an unknown error.
-      return DataStoreException(
-          AmplifyExceptionMessages.missingExceptionMessage,
-          recoverySuggestion:
-              AmplifyExceptionMessages.missingRecoverySuggestion,
-          underlyingException: e.toString());
-    }
-  }
-}
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/types/index.dart';
+import 'package:amplify_datastore/amplify_datastore.dart';
+import 'package:flutter/services.dart';
+import 'package:amplify_core/types/exception/AmplifyExceptionMessages.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+const MethodChannel _channel = MethodChannel('com.amazonaws.amplify/datastore');
+
+/// An implementation of [AmplifyDataStore] that uses method channels.
+class AmplifyDataStoreMethodChannel extends AmplifyDataStore {
+  var _allModelsStreamFromMethodChannel = null;
+
+  /// This method instantiates the native DataStore plugins with plugin
+  /// configurations. This needs to happen before Amplify.configure() can be
+  /// called.
+  @override
+  Future<void> configureDataStore(
+      {ModelProviderInterface modelProvider,
+      int syncInterval,
+      int syncMaxRecords,
+      int syncPageSize}) async {
+    try {
+      return await _channel
+          .invokeMethod('configureDataStore', <String, dynamic>{
+        'modelSchemas':
+            modelProvider.modelSchemas.map((schema) => schema.toMap()).toList(),
+        'modelProviderVersion': modelProvider.version,
+        'syncInterval': syncInterval,
+        'syncMaxRecords': syncMaxRecords,
+        'syncPageSize': syncPageSize
+      });
+    } on PlatformException catch (e) {
+      if (e.code == "AmplifyAlreadyConfiguredException") {
+        throw AmplifyAlreadyConfiguredException(
+            AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
+            recoverySuggestion:
+                AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion);
+      } else {
+        throw _deserializeException(e);
+      }
+    }
+  }
```

Comment:
Only actually touched this right?

Reply:
That's correct, converted the file from CRLF to LF to align with other files in the codebase. Github thought entire file got rewritten.